app/helpers/application_html_formatters_helper.rb

```diff
@@ -11,15 +11,51 @@ module ApplicationHTMLFormattersHelper
                                          HTML::Pipeline::RougeFilter
                                        ], DefaultPipelineOptions)
 
+  # List of video hosting site URLs to allow
+  VIDEO_URL_WHITELIST = Regexp.union(
+    /\A(?:https?:)?\/\/(?:www\.)?youtube\.com\//,
+    /\A(?:https?:)?\/\/(?:www\.)?youtu.be\//,
+    /\A(?:https?:)?\/\/(?:www\.)?vimeo\.com\//,
+    /\A(?:https?:)?\/\/(?:www\.)?vine\.co\//,
+    /\A(?:https?:)?\/\/(?:www\.)?instagram\.com\//,
+    /\A(?:https?:)?\/\/(?:www\.)?dailymotion\.com\//,
+    /\A(?:https?:)?\/\/(?:www\.)?youku\.com\//
+  )
+
+  # Transformer to whitelist iframes containing embedded video content
+  VIDEO_WHITELIST_TRANSFORMER = lambda do |env|
+    node, node_name = env[:node], env[:node_name]
+
+    return if env[:is_whitelisted] || !node.element?
+
+    return unless node_name == 'iframe'
+    return unless node['src'].match VIDEO_URL_WHITELIST
+
+    Sanitize.node!(node, elements: %w(iframe),
+                         attributes: {
+                           'iframe' => %w(allowfullscreen frameborder height src width)
```

Comment:
@jeremyyap small issue, I think we don't use `%w` in the project, can you change to sth like  `iframe: ['allowfullscreen']`? 

LGTM after this.